### PR TITLE
changing commandline arguments to be passed in a key value fashion

### PR DESCRIPTION
**Pull request by jerrypeng**
- changing commandline arguments to be passed in a key value fashion
- allow override config in heron_executor.py to actually override heron internal configs in HeronInstance

Splitting up PR: https://github.com/twitter/heron/pull/2468

### CODE REVIEW

**Comment by jerrypeng** [comment]
@srkukarni @maosongfu can you take a look at this. Thanks!

File: heron/instance/src/java/BUILD [hunk 1]
```diff
@@ -13,6 +13,7 @@ instance_deps_files =  \
         "//heron/common/src/java:config-java",
         "//heron/common/src/java:network-java",
         "//heron/common/src/java:utils-java",
+        "@commons_cli_commons_cli//jar"
```
Comment (srkukarni) [inline heron/instance/src/java/BUILD]: If we are introducing this dep, we need to add a shading rule for this.

Comment (jerrypeng) [reply heron/instance/src/java/BUILD]: I have added a rule to shade this dep

**Comment by objmagic** [comment]
have you changed Python instance as well?

**Comment by jerrypeng** [comment]
@objmagic I have not, but this change should not break it.  Feel free to work on changing Python Heron instance to accept arguments in this fashion if you want to.

**Comment by maosongfu** [comment]
@huijunw @objmagic We construct the arguments for heron-executor on aurora file, i.e. on the heron-cli side. We need to make sure a heron-cli will always deploy with the corresponding version of heron-core; otherwise, it will fail.

**Pull request merged**